CMakeLists.txt

```diff
@@ -0,0 +1,389 @@
+  if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")
+    set(PREBUILT_HOST_TOOLS ${PROJECT_SOURCE_DIR}/scripts/prebuilt)
+  endif()
+endif()
+
+if(NOT PREBUILT_HOST_TOOLS)
+  message(FATAL_ERROR "Building host tools not yet supported, please use PREBUILT_HOST_TOOLS")
+endif()
+
+include(cmake/extensions.cmake)
+include(cmake/kconfig.cmake)
+
+# set the variables VERSION_MAJOR, PATCHLEVEL and more
+include(cmake/version.cmake)
+
+set(SOC_NAME ${CONFIG_SOC})
+set(SOC_SERIES ${CONFIG_SOC_SERIES})
+set(SOC_FAMILY ${CONFIG_SOC_FAMILY})
+
+if("${SOC_SERIES}" STREQUAL "")
+  set(SOC_PATH ${SOC_NAME})
+else()
+  set(SOC_PATH ${SOC_FAMILY}/${SOC_SERIES})
+endif()
+
+set(ARCH ${CONFIG_ARCH})
+set(BOARD_NAME ${CONFIG_BOARD})
+set(KERNEL_NAME ${CONFIG_KERNEL_BIN_NAME})
+
+set(KERNEL_ELF_NAME ${KERNEL_NAME}.elf)
+set(KERNEL_BIN_NAME ${KERNEL_NAME}.bin)
+set(KERNEL_HEX_NAME ${KERNEL_NAME}.hex)
+set(KERNEL_STAT_NAME ${KERNEL_NAME}.stat)
+
+include(dts/dts.cmake)
+
+include_directories(
+  kernel/include
+  arch/${ARCH}/include
+  arch/${ARCH}/soc/${SOC_PATH}
+  boards/${ARCH}/${BOARD_NAME}
+  include
+  lib/libc/minimal/include
+  ${PROJECT_BINARY_DIR}/include/generated
+  ${USERINCLUDE}
+  ${STDINCLUDE}
+)
+
+add_definitions(
+  -DKERNEL
+  -D__ZEPHYR__=1
+  -include ${AUTOCONF_H}
+  -D_FORTIFY_SOURCE=2
+)
+
+add_compile_options(
+  -Os # TODO: allow config to override this
+  -g # TODO: build configuration enough?
+  -Wall
+  -Wformat
+  -Wformat-security
+  -Wno-format-zero-length
+  -Wno-main
+  -ffreestanding
+)
+
+add_compile_options(
+  $<$<COMPILE_LANGUAGE:C>:-std=c99>
+
+  $<$<COMPILE_LANGUAGE:CXX>:-std=c++11>
+  $<$<COMPILE_LANGUAGE:CXX>:-fcheck-new>
+  $<$<COMPILE_LANGUAGE:CXX>:-ffunction-sections>
+  $<$<COMPILE_LANGUAGE:CXX>:-fdata-sections>
+  $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>
+  $<$<COMPILE_LANGUAGE:CXX>:-fno-exceptions>
+
+  $<$<COMPILE_LANGUAGE:ASM>:-xassembler-with-cpp>
+  $<$<COMPILE_LANGUAGE:ASM>:-D_ASMLANGUAGE>
+)
+
+ld_option(-nostartfiles)
+ld_option(-nodefaultlibs)
+ld_option(-nostdlib)
+ld_option(-static)
+
+if(CONFIG_READABLE_ASM)
+  cc_option(-fno-reorder-blocks)
+  cc_option(-fno-ipa-cp-clone)
+  cc_option(-fno-partial-inlining)
+endif()
+
+cc_option(-fno-asynchronous-unwind-tables)
+
+if(CONFIG_STACK_CANARIES)
+  cc_option(-fstack-protector-all)
+else()
+  cc_option(-fno-stack-protector)
+endif()
+
+if(CONFIG_OVERRIDE_FRAME_POINTER_DEFAULT)
+  if(CONFIG_OMIT_FRAME_POINTER)
+    cc_option(-fomit-frame-pointer)
+  else()
+    cc_option(-fno-omit-frame-pointer)
+  endif()
+endif()
+
+# TODO: Handle default to Os
+add_compile_options(${CONFIG_COMPILER_OPT})
+
+# TODO: Include arch compiler options at this point.
+
+if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+   cc_disable_warning(no-unknown-warning-option)
+   cc_disable_warning(unused-variable)
+   cc_disable_warning(format-invalid-specifier)
+   cc_disable_warning(gnu)
+   # comparison of unsigned expression < 0 is always false
+   cc_disable_warning(tautological-compare)
```

Comment:
Although it can't be seen from the github snippet, it is also only disabled for clang builds in CMake. So Kbuild and CMake are behaving the same here.